perceval/components/source.py

```diff
@@ -21,66 +21,129 @@
+                 purity: float = None,
                  indistinguishability: float = 1,
-                 indistinguishability_model: str = "homv",  # Literal["homv", "linear"]
+                 indistinguishability_model: Literal["homv", "linear"] = "homv",
+                 overall_transmission: float = 1,
                  context: Dict = None) -> None:
+        r"""Definition of a source
+
+        :param brightness: the probability per laser pulse to emmit at least one photon. Indenpendent of all losses.
+        :param multiphoton_component: second order intensity autocorrelation at zero time delay :math:`g^{(2)}(0)`
+        :param multiphoton_model: `distinguishable` if additional photons are distinguishable, `indistinguishable`
+          otherwise
+        :param purity: preserved for back-compatibility if multiphoton_model is not set.`
+        :param indistinguishability: indistinguishability parameter as defined by `indistinguishability_model`
+        :param indistinguishability_model: `homv` defines indistinguishability as 2-photon wavepacket overlap,
+            `linear` defines indistinguishability as ratio of indistinguishable photons
+        :param overall_transmission: transmission of the optical system.
+        :param context: gives a local context for source specific features, like `discernability_tag`
+        """
+        if multiphoton_component is None:
+            if purity is None:
+                multiphoton_component = 0
+            else:
+                p2 = brightness * (1 - purity)
+                p1 = brightness - p2
+                multiphoton_component = 2 * p2 / (p1 + 2 * p2) ** 2
+        else:
+            assert purity is None, "cannot set both purity and multiphoton_component"
+        assert brightness * multiphoton_component <= 0.5, "brightness * g2 higher than 0.5 can not be computed for now"
         self.brightness = brightness
-        self.purity = purity
-        self._purity_model = purity_model
+        self.overall_transmission = overall_transmission
+        self.multiphoton_component = multiphoton_component
+        self._multiphoton_model = multiphoton_model
+        assert self._multiphoton_model in ["distinguishable", "indistinguishable"], "invalid value for purity_model"
         self.indistinguishability = indistinguishability
         self._indistinguishability_model = indistinguishability_model
         assert self._indistinguishability_model in ["homv", "linear"], "invalid value for indistinguishability_model"
         self._context = context or {}
         if "discernability_tag" not in self._context:
             self._context["discernability_tag"] = 1
 
-    def probability_distribution(self) -> SVDistribution:
-        r""":return: SVDistribution on 1 mode associated to the source
+    def probability_distribution(self):
+        r"""returns SVDistribution on 1 mode associated to the source
```

Comment:
This method is far too long to be readable. It should be split in logical chunks.

Reply:
Actually, @Alex-PLACET made the same kind of review in the other PR. His comments should be taken into account too.